extendeddatasquare.go

```diff
@@ -170,8 +170,8 @@ func (eds *ExtendedDataSquare) erasureExtendCol(codec Codec, i uint) error {
 }
 
 func (eds *ExtendedDataSquare) deepCopy(codec Codec) (ExtendedDataSquare, error) {
```

Comment:
[General comment] The documentation for the rsmt2d package functions is generally insufficient (or there is none in many cases). It would greatly benefit from improvements in this area.